pcsx2/GS/Renderers/HW/GSTextureCache.cpp

```diff
@@ -1047,6 +1047,68 @@ void GSTextureCache::InvalidateLocalMem(const GSOffset& off, const GSVector4i& r
 	// TODO: ds
 }
 
+bool GSTextureCache::Move(u32 SBP, u32 SBW, u32 SPSM, int sx, int sy, u32 DBP, u32 DBW, u32 DPSM, int dx, int dy, int w, int h)
+{
+	// TODO: In theory we could do channel swapping on the GPU, but we haven't found anything which needs it so far.
+	// Same with SBP == DBP, but this behavior could change based on direction?
+	if (SPSM != DPSM || SBP == DBP)
+	{
+		DevCon.Warning("Skipping HW move from 0x%X to 0x%X with SPSM=%u DPSM=%u", SBP, DBP, SPSM, DPSM);
+		return false;
+	}
+
+	// DX11/12 is a bit lame and can't partial copy depth targets. We could do this with a blit instead,
+	// but so far haven't seen anything which needs it.
+	if (GSConfig.Renderer == GSRendererType::DX11 || GSConfig.Renderer == GSRendererType::DX12)
+	{
+		if (GSLocalMemory::m_psm[SPSM].depth || GSLocalMemory::m_psm[DPSM].depth)
+			return false;
+	}
+
+	// Look for an exact match on the targets.
+	GSTextureCache::Target* src = GetExactTarget(SBP, SBW, SPSM);
+	GSTextureCache::Target* dst = GetExactTarget(DBP, DBW, DPSM);
```

Comment:
Now that I'm thinking of it, it dst does not exist in cache you can just copy src, change the BP/BW fields and add it to the relevant list. Also it might be needed to check that no overlap exist between the src and dst GS memory regions...but... that would be an evil move.

Reply:
Good suggestion, but for the record, I wanted to keep this change as small as possible in case of regressions. Could be worth doing in the future, but I don't think there's any cases where it does happen currently (at least in the ones I tested, it was moves *inside* an existing target).

Also keep in mind that the missing-from-cache cache would only happen once, with the previous frame's target being around for the next frame. So we'd probably want to drop the old target in such a case.